models/models_test.go

```diff
@@ -47,318 +48,322 @@ func (_f MockUtilsIam) DeleteCertificate(certName string) error {
+				}
 
-			data := r.Data.(*cloudfront.ListDistributionsOutput)
-			data.DistributionList = &cloudfront.DistributionList{
-				IsTruncated: aws.Bool(false),
-				Items:       list,
+				data := r.Data.(*cloudfront.ListDistributionsOutput)
+				data.DistributionList = &cloudfront.DistributionList{
+					IsTruncated: aws.Bool(false),
+					Items:       list,
+				}
 			}
-		}
-	})
-
-	mui := new(MockUtilsIam)
-	mui.Settings = settings
-	mui.Service = fakeiam
+		})
 
-	// expect the orphaned certs to be deleted
-	mui.On("DeleteCertificate", "some-orphaned-cert").Return(nil)
-	mui.On("DeleteCertificate", "some-other-orphaned-cert").Return(nil)
+		mui := new(MockUtilsIam)
+		mui.Settings = settings
+		mui.Service = fakeiam
 
-	m := models.NewManager(
-		logger,
-		mui,
-		&utils.Distribution{settings, fakecf},
-		settings,
-		&gorm.DB{},
-	)
+		// expect the orphaned certs to be deleted
+		mui.On("DeleteCertificate", "some-orphaned-cert").Return(nil)
+		mui.On("DeleteCertificate", "some-other-orphaned-cert").Return(nil)
 
-	//run the test
-	m.DeleteOrphanedCerts()
+		m := models.NewManager(
+			logger,
+			mui,
+			&utils.Distribution{settings, fakecf},
+			settings,
+			&gorm.DB{},
+		)
 
-	//check our expectations
-	mui.AssertExpectations(t)
-}
+		//run the test
+		m.DeleteOrphanedCerts()
 
-func TestDeleteOrphanedCertsDeleteFails(t *testing.T) {
-	logger := lager.NewLogger("cdn-cron-test")
-	logOutput := bytes.NewBuffer([]byte{})
-	logger.RegisterSink(lager.NewWriterSink(logOutput, lager.ERROR))
-
-	settings, _ := config.NewSettings()
-	session := session.New(nil)
-
-	//mock out the aws call to return a certificate to delete
-	fakeiam := iam.New(session)
-	fakeiam.Handlers.Clear()
-	fakeiam.Handlers.Send.PushBack(func(r *request.Request) {
-		//t.Log(r.Operation.Name)
-		switch r.Operation.Name {
-		case "ListServerCertificates":
-			old := time.Now().AddDate(0, 0, -2)
-
-			list := []*iam.ServerCertificateMetadata{
-				&iam.ServerCertificateMetadata{
-					Arn: aws.String("some-orphaned-cert"),
-					ServerCertificateName: aws.String("some-orphaned-cert"),
-					ServerCertificateId:   aws.String("this-cert-should-be-deleted"),
-					UploadDate:            &old,
-				},
-			}
-			data := r.Data.(*iam.ListServerCertificatesOutput)
-			data.IsTruncated = aws.Bool(false)
-			data.ServerCertificateMetadataList = list
-		}
+		//check our expectations
+		mui.AssertExpectations(GinkgoT())
 	})
 
-	//mock out the aws call to return a fixed list of distributions
-	fakecf := cloudfront.New(session)
-	fakecf.Handlers.Clear()
-	fakecf.Handlers.Send.PushBack(func(r *request.Request) {
-		//t.Log(r.Operation.Name)
-		switch r.Operation.Name {
-		case "ListDistributions2016_11_25":
-			list := []*cloudfront.DistributionSummary{}
-			data := r.Data.(*cloudfront.ListDistributionsOutput)
-			data.DistributionList = &cloudfront.DistributionList{
-				IsTruncated: aws.Bool(false),
-				Items:       list,
+	It("should fail to delete orphaned certs", func() {
```

Comment:
Perhaps "should handle AWS certificate deletion failure gracefully"?

Reply:
Resolved in 5a1b1f8